### PR TITLE
Fix pull engine runs before creation tx commit

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

File: execution/test_utils.go [hunk 1]
```diff
@@ -408,17 +405,30 @@ func ReadAllWithCount(ctx context.Context, stateStorage storage.Storage, stream
...
+	tx := stateStorage.BeginTransaction()
+	stream, execOutput, err := node.Get(storage.InjectStateTransaction(context.Background(), tx), variables, streamID)
 	if err != nil {
-		t.Fatal("couldn't open in-memory badger database: ", err)
+		t.Fatal(err)
+	}
+	for _, task := range execOutput.TasksToRun {
+		go task()
+	}
+	if err := tx.Commit(); err != nil {
```
Comment (woojtek) [inline execution/test_utils.go:430]: newline here please

File: execution/execution.go [hunk 2]
```diff
@@ -11,10 +11,13 @@ import (
 	"github.com/pkg/errors"
 )
 
+type Task func() error
+
 // This struct represents additional metadata to be returned with Get() and used recursively (like WatermarkSource)
 type ExecutionOutput struct {
```
Comment (woojtek) [inline execution/execution.go:17]: actually, after submitting approve, I would like You to write proper comments to `e v e r y` single field of ExecutionOutput as it grows!

Comment (cube2222) [reply execution/execution.go:17]: sure

File: storage/excel/datasource.go [hunk 3]
```diff
@@ -110,13 +110,13 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
...
-		log.Println("worker start")
-		rs.RunWorker(ctx)
-		log.Println("worker done")
-	}()
-
-	return rs, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator(), map[string]execution.ShuffleData{}), nil
+	return rs,
+		execution.NewExecutionOutput(
+			execution.NewZeroWatermarkGenerator(),
+			map[string]execution.ShuffleData{},
```
Comment (JasiekChomiak) [inline storage/excel/datasource.go:116]: I don't know which one is more elegant, but this `map[string]execution.ShuffleData{}` appears in multiple places - maybe just define `EmptyShuffleData` and use it

File: storage/json/datasource.go [hunk 4]
```diff
@@ -80,13 +80,13 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
...
-		log.Println("worker start")
-		rs.RunWorker(ctx)
-		log.Println("worker done")
-	}()
-
-	return rs, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator(), map[string]execution.ShuffleData{}), nil
+	return rs,
+		execution.NewExecutionOutput(
+			execution.NewZeroWatermarkGenerator(),
+			map[string]execution.ShuffleData{},
```
Comment (JasiekChomiak) [inline storage/json/datasource.go:86]: As above

File: storage/kafka/datasource.go [hunk 5]
```diff
@@ -145,12 +145,13 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
...
-	go func() {
-		rs.RunWorker(ctx)
-		log.Println("worker done")
-	}()
-
-	return rs, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator(), map[string]execution.ShuffleData{}), nil
+	return rs,
+		execution.NewExecutionOutput(
+			execution.NewZeroWatermarkGenerator(),
+			map[string]execution.ShuffleData{},
```
Comment (JasiekChomiak) [inline storage/kafka/datasource.go:151]: As above

File: storage/redis/datasource.go [hunk 6]
```diff
@@ -121,13 +121,13 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
...
-		log.Println("worker start")
-		rs.RunWorker(ctx)
-		log.Println("worker done")
-	}()
-
-	return rs, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator(), map[string]execution.ShuffleData{}), nil
+	return rs,
+		execution.NewExecutionOutput(
+			execution.NewZeroWatermarkGenerator(),
+			map[string]execution.ShuffleData{},
```
Comment (JasiekChomiak) [inline storage/redis/datasource.go:127]: As above

File: storage/sql/template.go [hunk 7]
```diff
@@ -135,13 +135,13 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
...
-		log.Println("worker start")
-		rs.RunWorker(ctx)
-		log.Println("worker done")
-	}()
-
-	return rs, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator(), map[string]execution.ShuffleData{}), nil
+	return rs,
+		execution.NewExecutionOutput(
+			execution.NewZeroWatermarkGenerator(),
+			map[string]execution.ShuffleData{},
```
Comment (JasiekChomiak) [inline storage/sql/template.go:141]: As above

**Pull request merged**